### PR TITLE
MM-22471 -  Implement Product Analytics to measure threads, posts and replies

### PR DESCRIPTION
**Pull request by reflog**
#### Summary
added additional analytics fields

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-22471

### CODE REVIEW

File: src/client/client4.ts [hunk 1]
```diff
@@ -1618,10 +1618,11 @@ export default class Client4 {
     // Post Routes
 
     createPost = async (post: Post) => {
-        this.trackEvent('api', 'api_posts_create', {channel_id: post.channel_id});
+        const analyticsData = {channel_id: post.channel_id, post_id: post.id, user_actual_id: post.user_id, root_id: post.root_id};
```
Comment (mickmister) [inline src/client/client4.ts:1621]: What's the background on `user_actual_id`?

Comment (reflog) [reply src/client/client4.ts:1621]: I think Chen or @adamjclarkson would be a better candidate for the answer :)

**Comment by amyblais** [comment]
@reflog Will this go through QA or is this good to merge?

**Comment by reflog** [comment]
> @reflog Will this go through QA or is this good to merge?

I think it's good to merge,also, you can't test analytics easily on spinwicks

File: src/client/client4.ts (hunk 1 above)
Comment (adamjclarkson) [reply src/client/client4.ts:1621]: Not sure what the background is, but I know `user_id` is the server id and `user_actual_id` is user id.

**Comment by saturninoabril** [comment]
Yes, good to merge. Analytics/diagnostics test will be handled separately in after merged.

**Comment by reflog** [comment]
@amyblais - you've remove PM review label so now I cannot merge

**Comment by amyblais** [comment]
@reflog Ok, added the label back (Adam has reviewed this)

**Comment by reflog** [comment]
@amyblais merging is still blocked :( @metanerd any idea why?

**Comment by metanerd** [comment]
@reflog it was set on pending, because of the `1:PM Review` label.

**Comment by reflog** [comment]
@metanerd thanks!

**Comment by adamjclarkson** [comment]
@enelson720 can you share here what you found while testing this one out?

**Comment by enelson720** [comment]
@adamjclarkson 

> when you create the root message you just get api_posts_create with no post_id, then if you reply it sends two events api_posts_create with a root_id and api_posts_replied with the same root_id. 

The root "post_id" should be generated from the original post and propagated to all subsequent responses and reactions to that post. Right now it is the reply that generates the unique post_id. The flow should be as follows:

User 1 posts

- post_id = "123"
- event_name = "api_posts_create"
- post_id should be populated with a unique identifier for this specific post)

User 2 replies to User 1's post
- post_id = "123"
- event_name = "api_posts_replied"
- post_id should match User 1's post_id

Still need me to provide additional context?

**Comment by adamjclarkson** [comment]
Thx @enelson720 bug was identified and fixed here: https://github.com/mattermost/mattermost-redux/pull/1148

**Pull request merged**